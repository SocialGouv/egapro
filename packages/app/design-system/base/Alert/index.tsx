import React from "react"
import clsx from "clsx"

interface AlertPropsCommon {
  type?: "error" | "warning" | "success" | "info"
}

interface AlertSM extends AlertPropsCommon {
  description: string
  size?: "sm"
  title?: string
}

interface AlertMD extends AlertPropsCommon {
  description?: string
  size?: "md"
  title: string
}

type AlertProps = AlertSM | AlertMD

const Alert = ({
  type = "info",
  size = "md",
  title,
  description,
  ...rest
}: AlertProps) => {
  return (
    <div
      role="alert"
      className={clsx(
        "fr-alert",
        type === "error" && "fr-alert--error",
        type === "success" && "fr-alert--success",
        type === "info" && "fr-alert--info",
        type === "warning" && "fr-alert--warning",
        size === "sm" && "fr-alert--sm"
      )}
      {...rest}
    >
      {size === "md" && <p className="fr-alert__title">{title}</p>}
      {description && <p>{description}</p>}
    </div>
  )
}

export default Alert
