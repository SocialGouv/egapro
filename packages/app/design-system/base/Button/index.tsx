import React from "react"
import { buttonStyles, buttonStylesProps } from "../../utils/button-styles"

export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> &
  buttonStylesProps & {
    label: string
    isDisabled?: boolean
  }

const Button = ({ variant, size, label, iconLeft, iconRight, iconOnly, isDisabled, ...rest }: ButtonProps) => {
  return (
    <button
      className={buttonStyles(variant, size, iconLeft, iconRight, iconOnly)}
      disabled={isDisabled}
      title={iconOnly && label}
      {...rest}
    >
      {label}
    </button>
  )
}

export default Button
