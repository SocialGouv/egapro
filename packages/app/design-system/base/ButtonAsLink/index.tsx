import React from "react"
import { buttonStyles, buttonStylesProps } from "../../utils/button-styles"

export type ButtonProps = React.AnchorHTMLAttributes<HTMLAnchorElement> &
  buttonStylesProps & {
    label: string
  }

const ButtonAsLink = ({ variant, size, label, iconLeft, iconRight, iconOnly, ...rest }: ButtonProps) => {
  return (
    <a className={buttonStyles(variant, size, iconLeft, iconRight, iconOnly)} title={iconOnly && label} {...rest}>
      {label}
    </a>
  )
}

export default ButtonAsLink
