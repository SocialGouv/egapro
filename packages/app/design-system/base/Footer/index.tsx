import React, { FunctionComponent } from "react"

export type FooterProps = {}

const Footer: FunctionComponent<FooterProps> = ({ children }) => {
  return (
    <footer className="fr-footer" role="contentinfo" id="footer">
      <div className="fr-container">{children}</div>
    </footer>
  )
}

export default Footer
