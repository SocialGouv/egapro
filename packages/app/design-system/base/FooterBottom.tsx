import React, { FunctionComponent } from "react"

export type FooterBottomProps = {}

export const FooterBottom: FunctionComponent<FooterBottomProps> = ({ children }) => {
  return (
    <div className="fr-footer__bottom">
      <ul className="fr-footer__bottom-list">{children}</ul>
      <div className="fr-footer__bottom-copy">
        <p>
          Sauf mention contraire, tous les contenus de ce site sont sous{" "}
          <a href="https://github.com/etalab/licence-ouverte/blob/master/LO.md" target="_blank" rel="noreferrer">
            licence etalab-2.0
          </a>
        </p>
      </div>
    </div>
  )
}
