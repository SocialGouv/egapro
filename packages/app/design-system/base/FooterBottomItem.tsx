import React, { FunctionComponent } from "react"

export type FooterBottomItemProps = {}

export const FooterBottomItem: FunctionComponent<FooterBottomItemProps> = ({ children }) => {
  return <li className="fr-footer__bottom-item">{children}</li>
}
