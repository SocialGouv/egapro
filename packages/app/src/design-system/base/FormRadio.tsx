import clsx from "clsx";
import type { PropsWithChildren, ReactNode } from "react";
import React, { Children } from "react";

import type { AuthorizedChildType } from "../utils/compatible-components";
import { compatibleComponents } from "../utils/compatible-components";

export type FormRadioGroupProps = PropsWithChildren<
  React.ReactHTMLElement<HTMLDivElement> & {
    isDisabled?: boolean;
    isError?: boolean;
    isValid?: boolean;
  }
>;

export const FormRadioGroup = ({ children, isError, isValid, isDisabled }: FormRadioGroupProps) => {
  const arrayOfChildren = Children.toArray(children);
  compatibleComponents(["FormRadioInput", "FormRadioLegend", "ValidationMessage"], arrayOfChildren);
  const legend = arrayOfChildren.filter(child => (child as AuthorizedChildType).type.name === "FormRadioLegend");
  const content = arrayOfChildren.filter(child => (child as AuthorizedChildType).type.name === "FormRadioInput");
  const validation = arrayOfChildren.filter(child => (child as AuthorizedChildType).type.name === "ValidationMessage");

  return (
    <div className="fr-form-group">
      <fieldset
        className={clsx("fr-fieldset", isError && "fr-fieldset--error", isValid && "fr-fieldset--valid")}
        disabled={isDisabled}
        role="group"
      >
        {legend}
        <div className="fr-fieldset__content">{content}</div>
        {validation.length > 0 && validation}
      </fieldset>
    </div>
  );
};

FormRadioGroup.Legend = function FormRadioLegend({ children, id }: { children: ReactNode; id: string }) {
  return (
    <legend className="fr-fieldset__legend fr-text--regular" id={id}>
      {children}
    </legend>
  );
};

FormRadioGroup.Input = function FormRadioInput({
  children,
  id,
  size,
}: {
  children: ReactNode;
  id: string;
  size?: "md" | "sm";
}) {
  const sizeClass = size ? size : "md";
  return (
    <div className={clsx("fr-radio-group", `fr-radio-group--${sizeClass}`)}>
      <input id={id} type="radio" />
      <label className="fr-label" htmlFor={id}>
        {children}
      </label>
    </div>
  );
};

FormRadioGroup.ValidationMessage = function ValidationMessage({
  children,
  isError,
  isValid,
}: {
  children: ReactNode;
  isError?: boolean;
  isValid?: boolean;
}) {
  return (
    <p
      id="radio-valid-desc-valid"
      className={clsx("fr-fieldset", isError && "fr-error-text", isValid && "fr-valid-text")}
    >
      {children}
    </p>
  );
};
