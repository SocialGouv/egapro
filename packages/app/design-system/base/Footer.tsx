import React from "react"

import { FooterBody } from "./FooterBody"
import { FooterBodyItem } from "./FooterBodyItem"
import { FooterBottom } from "./FooterBottom"
import { FooterBottomItem } from "./FooterBottomItem"
import { FooterBottomLink } from "./FooterBottomLink"
import { FooterContentLink } from "./FooterContentLink"
import { Logo } from "./Logo"

export const Footer = () => {
  return (
    <footer className="fr-footer" role="contentinfo" id="footer">
      <div className="fr-container">
        <FooterBody
          logo={
            <a href="#">
              <Logo />
            </a>
          }
          description="Représentation Équilibrée a été développé par les équipes de la fabrique numérique des ministères sociaux."
          items={
            <>
              <FooterBodyItem>
                <FooterContentLink href="https://legifrance.gouv.fr" target="_blank" rel="noreferrer">
                  legifrance.gouv.fr
                </FooterContentLink>
              </FooterBodyItem>
              <FooterBodyItem>
                <FooterContentLink href="https://legifrance.gouv.fr" target="_blank" rel="noreferrer">
                  legifrance.gouv.fr
                </FooterContentLink>
              </FooterBodyItem>
            </>
          }
        />
        <FooterBottom>
          <>
            <FooterBottomItem>
              <FooterBottomLink href="https://legifrance.gouv.fr">Plan du site</FooterBottomLink>
            </FooterBottomItem>
            <FooterBottomItem>
              <FooterBottomLink href="https://legifrance.gouv.fr">Mentions légales</FooterBottomLink>
            </FooterBottomItem>
          </>
        </FooterBottom>
      </div>
    </footer>
  )
}
