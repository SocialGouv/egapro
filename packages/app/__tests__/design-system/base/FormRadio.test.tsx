import { render } from "@testing-library/react";
import { FormRadioGroup } from "@/design-system/base/FormRadio";

test("should match snapshot", () => {
  const view = render(
    <FormRadioGroup isValid>
      <FormRadioGroup.Legend id="legendId">Légende</FormRadioGroup.Legend>
      <FormRadioGroup.Input id="inputId1" size="sm">
        radio 1
      </FormRadioGroup.Input>
      <FormRadioGroup.Input id="inputId2" size="sm">
        radio 2
      </FormRadioGroup.Input>
      <FormRadioGroup.Input id="inputId3" size="sm">
        radio 3
      </FormRadioGroup.Input>
      <FormRadioGroup.ValidationMessage isValid>Validé</FormRadioGroup.ValidationMessage>
    </FormRadioGroup>,
  );
  expect(view).toMatchSnapshot();
});
