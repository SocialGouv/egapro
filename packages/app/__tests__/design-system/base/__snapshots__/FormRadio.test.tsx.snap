// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fr-form-group"
      >
        <fieldset
          class="fr-fieldset fr-fieldset--valid"
          role="group"
        >
          <legend
            class="fr-fieldset__legend fr-text--regular"
            id="legendId"
          >
            Légende
          </legend>
          <div
            class="fr-fieldset__content"
          >
            <div
              class="fr-radio-group fr-radio-group--sm"
            >
              <input
                id="inputId1"
                type="radio"
              />
              <label
                class="fr-label"
                for="inputId1"
              >
                radio 1
              </label>
            </div>
            <div
              class="fr-radio-group fr-radio-group--sm"
            >
              <input
                id="inputId2"
                type="radio"
              />
              <label
                class="fr-label"
                for="inputId2"
              >
                radio 2
              </label>
            </div>
            <div
              class="fr-radio-group fr-radio-group--sm"
            >
              <input
                id="inputId3"
                type="radio"
              />
              <label
                class="fr-label"
                for="inputId3"
              >
                radio 3
              </label>
            </div>
          </div>
          <p
            class="fr-fieldset fr-valid-text"
            id="radio-valid-desc-valid"
          >
            Validé
          </p>
        </fieldset>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fr-form-group"
    >
      <fieldset
        class="fr-fieldset fr-fieldset--valid"
        role="group"
      >
        <legend
          class="fr-fieldset__legend fr-text--regular"
          id="legendId"
        >
          Légende
        </legend>
        <div
          class="fr-fieldset__content"
        >
          <div
            class="fr-radio-group fr-radio-group--sm"
          >
            <input
              id="inputId1"
              type="radio"
            />
            <label
              class="fr-label"
              for="inputId1"
            >
              radio 1
            </label>
          </div>
          <div
            class="fr-radio-group fr-radio-group--sm"
          >
            <input
              id="inputId2"
              type="radio"
            />
            <label
              class="fr-label"
              for="inputId2"
            >
              radio 2
            </label>
          </div>
          <div
            class="fr-radio-group fr-radio-group--sm"
          >
            <input
              id="inputId3"
              type="radio"
            />
            <label
              class="fr-label"
              for="inputId3"
            >
              radio 3
            </label>
          </div>
        </div>
        <p
          class="fr-fieldset fr-valid-text"
          id="radio-valid-desc-valid"
        >
          Validé
        </p>
      </fieldset>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
